src/dotnet-restore/Program.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Tools.Restore
+{
+    public class Program
+    {
+        public static int Main(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            var app = new CommandLineApplication(false)
+            {
+                Name = "dotnet restore",
+                FullName = ".NET project dependency restorer",
+                Description = "Restores dependencies listed in project.json"
+            };
+            
+            app.OnExecute(() =>
+            {
+                try
+                {
+                    var projectRestoreResult = Dnx.RunRestore(args);
+
+                    var restoreTasks = GetRestoreTasks(args);
+
+                    if (!restoreTasks.Any()) return projectRestoreResult;
```

Comment:
You shouldn't really need this check here. Get rid of it and remove one branching from the code.